burn-core/src/nn/initializer.rs

```diff
@@ -0,0 +1,107 @@
+use burn_tensor::Shape;
+
+use crate::config::Config;
+use crate::tensor::backend::Backend;
+use crate::tensor::{Distribution, ElementConversion, Tensor};
+
+use crate as burn;
+
+#[derive(Config, Debug, PartialEq)]
+pub enum Initializer {
+    Uniform(f64, f64),
+    UniformDefault,
```

Comment:
Sounds good. I could not get the Elem type working because it requires backend. If you're referring to UniformDefault enum than I kinda agree but I don't have any better idea either.